packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -146,7 +146,12 @@ export class ApolloServerBase<
   // The constructor should be universal across all environments. All environment specific behavior should be set by adding or overriding methods
   constructor(config: Config<ContextFunctionParams>) {
     if (!config) throw new Error('ApolloServer requires options.');
-    this.config = config;
+
+    this.config = {
+      ...config,
+      nodeEnv: config.nodeEnv ?? process.env.NODE_ENV,
+    };
```

Comment:
OK so... my understanding.

The `nodeEnv` will be initialized here, either via a direct configuration, or via the NODE_ENV value.